statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -657,26 +961,42 @@ def acorr(self, nlags=None):
 
     def plot_acorr(self, nlags=10, linewidth=8):
         "Plot theoretical autocorrelation function"
-        plotting.plot_full_acorr(self.acorr(nlags=nlags), linewidth=linewidth)
+        plotting.plot_full_acorr(self.acorr(nlags=nlags), linewidth=linewidth,
+                                 names=self.model.endog_names)
 
-    def forecast(self, y, steps):
-        """Produce linear minimum MSE forecasts for desired number of steps
-        ahead, using prior values y
+    def forecast(self, y=None, steps=1):
+        """Minimum MSE forecasts for desired number of steps
 
         Parameters
         ----------
-        y : ndarray (p x k)
+        y : ndarray (k_ar x neqs)
+            Deprecated. The initial values to use for the forecasts. If None,
+            the last k_ar values of the original endogenous variables are
+            used. Use statsmodels.tsa.var.forecast instead if you need to
+            give y values.
         steps : int
+            The number of steps ahead to forecast.
 
         Returns
         -------
         forecasts : ndarray (steps x neqs)
-
-        Notes
-        -----
-        Lutkepohl pp 37-38
+            The forecasted values.
         """
-        return forecast(y, self.coefs, self.intercept, steps)
+        #TODO: give the math instead the reference in Notes
+        # Lutkepohl pp 37-38
+        k_ar = self.k_ar
+
+        if y is not None:
+            warn("The use of y is deprecated and will be removed in 0.6.0. "
+                 "If you want to give initial values use "
+                 "statsmodels.tsa.var.forecast.")
+        else:
+            y = self.model.Y[-k_ar:]
```

Comment:
do we need the model for the process?

process shouldn't depend on specific data